Makefile

```diff
@@ -15,59 +17,62 @@ DOCTESTS = $(addsuffix /doctest,$(filter-out meta,${PACKAGES}))
 # All linting targets.
 LINT = $(addsuffix /lint,${PACKAGES})
 
-# Targets to build a distribution for each package.
-dist: ${DISTRIBUTIONS}
-${DISTRIBUTIONS} : %/dist : %/setup.py
-	cd $* \
-	&& python setup.py bdist_wheel \
-	&& twine check dist/*
 
-# Targets to run the test suite for each package.
-tests : ${TESTS}
-${TESTS} : %/tests :
+install:  ## Set up the project for development
+	poetry install --all-extras
+	poetry run pre-commit install
+
+dist:  ## Build the python package
+	poetry build && poerry run twine check dist/*
+
+tests: ${TESTS}  ## Run tests for each package
+${TESTS}: %/tests:
 	poetry run pytest -v --cov=testcontainers.$* $*/tests
```

Comment:
Do we already have a plan what to do with our test coverage reports?

We could export it to [codecov](https://about.codecov.io/) and add a badge to our README.md.

Reply:
:wave: yep, I'd like that, but it needs @kiview's involvement as you need to install the GH app - I am missing some privileges that we'll be needing to install different apps.

We'll talk about this on slack I think.

Reply:
spit into #544

Reply:
*split